controlplane/kubeadm/internal/controllers/remediation.go

```diff
@@ -155,12 +156,20 @@ func (r *KubeadmControlPlaneReconciler) reconcileUnhealthyMachines(ctx context.C
 	// If the machine that is about to be deleted is the etcd leader, move it to the newest member available.
 	if controlPlane.IsEtcdManaged() {
 		etcdLeaderCandidate := controlPlane.HealthyMachines().Newest()
+		if etcdLeaderCandidate == nil {
+			err := errors.New("Failed to move etcd leadership: no healthy candidate machines found")
+			log.Error(err, "", "Machine")
```

Comment:
```suggestion
			err := errors.New("no healthy candidate machines found")
			log.Error(err, "Failed to move etcd leadership")
```

We are not wrapping the `Failed to ...` message into the error in the other cases. also `"Machine"` would require a value.

Reply:
👍 
I wonder if we should be wrapping that message into the error that appears in the condition - but you're right that it's inconsistent with the current approach.